### PR TITLE
feat(25055): Create a renderer and select component for aggregated topics

### PR DESCRIPTION
**Pull request by vanch3d**
See https://hivemq.kanbanize.com/ctrl_board/57/cards/25055/details/

This PR is part of the `bi-directional` epic and enable the frontend part of the feature. It consists of three elements:

- reorganise the `workspace` so that the page becomes the central point of exploiting `inbound` and `outbound` subscriptions
-  Introduce the notion of the `topic wheel` to act as a controller for aggregating `metadata` of the Edge topology and structuring its access for `outbound` subscriptions
- create a proof-of-concept for the interactive visualisation of the `topic wheel`

### Design
- The `workspace` has been refactored, enabling interaction with the `Edge` node and its side panel configuration. The `Edge` node is now a functional node similar to the other elements in the graph
- The `Edge` side panel contains the new `topic wheel` component, as a visualiser of the data domain aggregated from the `inward` subscription
- The `topic wheel` is built on top of a "sunburst" chart, rendering the combined topics of the topology into a planar, dynamic visualisation:
  - Concentric "rings" materialise the segment of the combined topic tree, from the root
  - "Sectors" materialise each topic by its segment, from the root to the leaf
  - the width of a given "sector" is based on dynamic metrics, currently set to the count of individual subscriptions
- Selecting a topic in the wheel opens a secondary panel with the data model(s) associated with the topic. 
- Wildcard in topics (mostly the `#`) opens a separate panel, allowing users to sample topics from the MQTT traffic over a certain period. 

### Out-of-scope
- The `topic wheel` is based on the `nivo` component (see https://nivo.rocks/sunburst/) but has significant accessibility issues that can only be fixed on the source (e.g. aria roles and zIndex for segments, aria-label for images, ...). This will be dealt with on a dedicated PR to the library
- The `metadata` collected for the topics are mock models and will be dealt with in further tickets (both backend and frontend)
- The `topic wheel` has `dev-only` elements that are only useful for PoC and prototyping (e.g. switch to tree view, UNS). They will be removed in the cleanup ticket
- As with other parts of the `Edge` frontend, the `topic wheel` will have scalability issues due to the breadth and depth of the "topic tree". There are several tools that could be used to alleviate the usability of the component (e.g. dynamic drill down, filtering); they will be explored in subsequent tickets.  
- Extracting and combining the topics is done in a very brute-force way, lacking proper handling of origin and UNS. The stratification routine will be improved in a dedicated ticket.

### After
screenshot-localhost_3000-2024 08 12-18_20_48

screenshot-localhost_3000-2024 08 12-18_20_58

screenshot-localhost_3000-2024 08 12-18_21_25

### CODE REVIEW

File: hivemq-edge/src/frontend/src/locales/en/translation.json [hunk 1]
```diff
@@ -738,6 +740,25 @@
...
+    },
+    "topicWheel": {
+      "control": {
+        "treeview": "Topic Tree View",
+        "sunburst": "Topic Wheel",
+        "uns": "UNS",
+        "origin": "Adapter"
+      },
+      "topicSampler": {
+        "title": "The device is using a wildcard to subscribe to topics, they are therefore not known at configuration.",
```
Comment (h2xd) [inline hivemq-edge/src/frontend/src/locales/en/translation.json:752]:
```suggestion
        "title": "The device is using a wildcard to subscribe to topics, they are therefore not known during it's configuration.",
```

File: hivemq-edge/src/frontend/src/modules/Workspace/components/topics/TopicSampler.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      <CardHeader as={HStack} justifyContent="space-between">
+        <Heading size="md">{topic}</Heading>
+      </CardHeader>
+      <CardBody data-testid="sampler-warning">
+        <Text>{t('workspace.topicWheel.topicSampler.title')} </Text>
+        <Text>{t('workspace.topicWheel.topicSampler.description')} </Text>
+      </CardBody>
+      <CardFooter>
+        <Button
+          onClick={() => topicSampler.mutateAsync({ adapter: 'sss', topic: 'ss' })}
```
Comment (h2xd) [inline hivemq-edge/src/frontend/src/modules/Workspace/components/topics/TopicSampler.tsx:25]: question(non-blocking): Wow that's a lot of `s` characters. Hope it's not an Snake adapter of some sort. Is this expected to have that may `s`'s?

File: hivemq-edge/src/frontend/src/modules/Workspace/components/topics/TreeViewChart.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  onSelect?: (topic: string) => void
+}
+
+const TreeViewChart: FC<TreeViewProps> = ({ data }) => {
+  return (
+    <ReactTreeView
+      data={flattenTree(data)}
+      nodeRenderer={({ element, getNodeProps, level, isBranch, isExpanded }) => {
+        return (
+          <HStack {...getNodeProps()} marginLeft={8 * (level - 1)}>
```
Comment (h2xd) [inline hivemq-edge/src/frontend/src/modules/Workspace/components/topics/TreeViewChart.tsx:20]: question(non-blocking): this `8` seems to be a magic number to add in visual difference. Maybe worth extracting to give it a name for a later view.

`const visualTabulatorSize = 8`

```ts
<HStack {...getNodeProps()} marginLeft={visualTabulatorSize * (level - 1)}>
```

**Review by h2xd — APPROVED** [review]
LGTM! 🐊

**Comment by h2xd** [comment]
Looking so nice!

File: hivemq-edge/src/frontend/src/modules/Workspace/components/topics/TopicSampler.tsx (hunk 2 above)
Comment (vanch3d) [reply hivemq-edge/src/frontend/src/modules/Workspace/components/topics/TopicSampler.tsx:25]: I knew there was some placeholder left somewhere

**Pull request merged**